src/vm/errors/vm_exception.rs

```diff
@@ -11,42 +14,69 @@ pub struct VmException {
     inst_location: Option<Location>,
     inner_exc: VirtualMachineError,
     error_attr_value: Option<String>,
+    traceback: Option<String>,
 }
 
 impl VmException {
-    pub fn from_vm_error(runner: &CairoRunner, error: VirtualMachineError, pc: usize) -> Self {
-        let error_attr_value = get_error_attr_value(pc, runner);
+    pub fn from_vm_error(
+        runner: &CairoRunner,
+        vm: &VirtualMachine,
+        error: VirtualMachineError,
+    ) -> Self {
+        let error_attr_value = get_error_attr_value(vm.run_context.pc.offset, runner);
         VmException {
-            pc,
-            inst_location: get_location(&pc, runner),
+            pc: vm.run_context.pc.offset,
+            inst_location: get_location(vm.run_context.pc.offset, runner),
             inner_exc: error,
             error_attr_value,
+            traceback: get_traceback(vm, runner),
         }
     }
 }
 
 pub fn get_error_attr_value(pc: usize, runner: &CairoRunner) -> Option<String> {
+    let mut errors = String::new();
     for attribute in &runner.program.error_message_attributes {
         if attribute.start_pc <= pc && attribute.end_pc > pc {
-            return Some(format!("Error message: {}\n", attribute.value));
+            errors.push_str(&format!("Error message: {}\n", attribute.value));
         }
     }
-    None
+    (!errors.is_empty()).then(|| errors)
 }
 
-pub fn get_location(pc: &usize, runner: &CairoRunner) -> Option<Location> {
+pub fn get_location(pc: usize, runner: &CairoRunner) -> Option<Location> {
     runner
         .program
         .instruction_locations
         .as_ref()?
-        .get(pc)
+        .get(&pc)
```

Comment:
Why does `get` take a reference here?

Reply:
Its due to the `HashMap::get()` signature